backend/src/datasources/userDatasource.js

```diff
@@ -45,6 +49,20 @@ export class UserDataSource extends DataSource {
   }
 
   async getAll () {
+    this.users = []
+    const session = driver.session()
+    const txc = session.beginTransaction()
+
+    try {
+      const result = await txc.run('MATCH(u:User) RETURN u')
+      await txc.commit()
+      result.records.forEach(r => this.users.push(r.get(0).properties))
+    } catch (error) {
+      console.log(error)
+    } finally {
+      await session.close()
+    }
+
```

Comment:
This is dangerous 🔥  Your method prefix is "get" but has a side effect. This is dangerous because you're overwriting the `this.user` variable thus manipulating the state of the object. Also every other method need to call `getAll` before accessing `this.user` because otherwise it is not guaranteed to hold the latest data.
Also this behavior is not documented so if you miss calling the method before a `this.user` call your state can be broken.